identity/handler.go

```diff
@@ -146,6 +155,17 @@ type listIdentitiesParameters struct {
 //	  200: listIdentities
 //	  default: errorGeneric
 func (h *Handler) list(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	if identifier := r.URL.Query().Get("identifier"); identifier != "" {
+		i, err := h.r.PrivilegedIdentityPool().FindByCredentialsIdentifier(r.Context(), identifier)
+		if err != nil {
+			h.r.Writer().WriteError(w, r, err)
+			return
+		}
+
+		h.r.Writer().Write(w, r, i)
```

Comment:
Similarly, if no result could be found, it should just be an empty array, not the `not found` error.

Reply:
@jonas-jonas I thought of that. There's a unique index on the identifier so there can't be multiple identities. Hence the pagination parameters also are pointless imo.

A bit off-topic but an API returning array responses is generally bad imo. We should generally have an object response (since that's extensible) with an items property or something.

Reply:
> There's a unique index on the identifier so there can't be multiple identities. Hence the pagination parameters also are pointless imo.

I tend to agree, but just silently changing the response schema based on an optional query parameter like this is not great behavior and will confuse people, I'd prefer getting ahead of that and just treating the `identifier` param as if it was a filter on the result set.

Alternatively, we can introduce a new endpoint, but I am not sure if that's a better.

> A bit off-topic but an API returning array responses is generally bad imo. We should generally have an object response (since that's extensible) with an items property or something.

Sure, but the existing API will break here. So I guess if we ever make a breaking overhaul of the API design (again) we can tackle these things. For now, please stick to the existing design.

Reply:
done. just wrapped the identity in a slice